memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -369,15 +369,16 @@ object HistogramBuckets {
   }
 
   // NOTE: must point to u16/Short length prefix bytes
-  def apply(bucketsDef: Ptr.U8, formatCode: Byte): HistogramBuckets = formatCode match {
-    case HistFormat_Geometric_Delta  => geometric(UnsafeUtils.ZeroArray, bucketsDef.add(2).addr, false)
-    case HistFormat_Geometric1_Delta => geometric(UnsafeUtils.ZeroArray, bucketsDef.add(2).addr, true)
-    case HistFormat_Custom_Delta     => custom(UnsafeUtils.ZeroArray, bucketsDef.addr)
+  def apply(acc: MemoryReader, bucketsDef: Ptr.U8, formatCode: Byte): HistogramBuckets = formatCode match {
+    case HistFormat_Geometric_Delta  => geometric(acc.base, acc.baseOffset + bucketsDef.add(2).addr, false)
+    case HistFormat_Geometric1_Delta => geometric(acc.base, acc.baseOffset + bucketsDef.add(2).addr, true)
+    case HistFormat_Custom_Delta     => custom(acc.base, bucketsDef.addr)
```

Comment:
Why is it that here baseOffset is not needed but it is needed for geometric above?  I think this needs to be fixed to be consistent.  Custom is what most Prom-format histograms use.

Reply:
Good catch, I'll fix but need to discuss this with you.